lib/spack/spack/test/cmd/env.py

```diff
@@ -1013,6 +1009,40 @@ def test_bad_env_yaml_format_remove():
     assert badenv not in env("list")
 
 
+@pytest.mark.parametrize("answer", ["-y", ""])
+def test_multi_env_remove(mutable_mock_env_path, monkeypatch, answer):
+    """Test removal (or not) of a valid and invalid environment"""
+    resp = answer == "-y"
+    monkeypatch.setattr(tty, "get_yes_or_no", lambda prompt, default: resp)
+
+    environments = ["goodenv", "badenv"]
+    for e in environments:
+        env("create", e)
+
+    # Ensure the bad environment contains invalid yaml
+    filename = mutable_mock_env_path / environments[1] / "spack.yaml"
+    with open(filename, "w") as f:
+        f.write(
```

Comment:
With `Path` objects This:
```suggestion
    filename.write_text(
```
should work too.

Reply:
Ah, introduced in v3.5 and I see it basically uses the context manager.  Cool!